### DISCUSSION TITLE
SummaryRow and Hyperformula

### DISCUSSION THREAD

**Discussion by haraleib** [handsontable/hyperformula#691] [General]
I have noticed that when using columnSummary, there is no result calculated in the cell which should calculate some result for a column with formulas.

Do I miss something here if I work with Hyperformula?
As a workaround I could skip the Handsontable columnSummary and provide my own summary filled with formulas which calculate the sum of the column.

The seetings of interest for the hot and hf instances (for the last column there should be a sum calculated)
![image](https://user-images.githubusercontent.com/19574757/121061443-239de580-c7c4-11eb-9639-5a5fa3d73df6.png)

UI
![image](https://user-images.githubusercontent.com/19574757/121060841-70cd8780-c7c3-11eb-8919-efed5082db30.png)

**Comment by AMBudnik** [comment]
Hi @haraleib 

the `SummaryRow` in not a method name in Handsontable nor HyperFormula. Do you mean the `columnSummary` (https://handsontable.com/docs/9.0.0/demo-summary-calculations.html)? If you could share a demo that would definitely help me to solve the case.

**Reply by haraleib** [reply]
My bad. I have updated the title and description and changed it to columnSummary.

Here the jsfiddle example:
https://jsfiddle.net/awy9d43f/1/

**Comment by AMBudnik** [comment]
The summary calculations are not formulas, they won't appear in the formula stack. 

And there is a difference between the demo (calculates properly and returns `hot.getData()` as it should) and the images where the value in the last row is not calculated. Do you mean that previously in your code you have also used the `summaryRow` and replacing it with `columnSummary` fixed the result?

**Reply by haraleib** [reply]
> The summary calculations are not formulas, they won't appear in the formula stack.

Yes, I thought so, but I guess I can achieve the same result without the columnSummary configuration, if I provide a normal row instaed of the columnSummary row which calculates the result with formulas if it turns out that it is a bug.

I wanted to share some data of interest for you with the image I provided.
In my code I also use columnSummary. The summaryRow I used here in this discussion was a typo.

Nevertheless, in the  jsfiddle example I provided the result is not calculated for the first column (which consists of formulas). Do I make something wrong here?

**Reply by AMBudnik** [reply]
Sorry @haraleib I did not see that.

The Formulas and columnSummary collaboration is not yet supported in the Handsontable v9.0.0.

Here https://github.com/handsontable/handsontable/projects/2 is a full list of changes made for the plugin and in the `Rejected/On hold` column (where we also have columnSummary) we listed all the topics that had to be postponed. We will provide more changes and updates in the future.

**Reply by haraleib** [reply]
Ok, so everything is clear for me. Thank you :)

**Reply by AMBudnik** [reply]
You're welcome 👋 

If you'd have any questions feel free to ask.
HyperFormula is still a fresh thing, and we may miss some info in the docs or examples, so all the feedback is priceless.

**Comment by julianaiglsdorfer** [comment]
Hello,

i just want to ask you if this issue/problem is still on your roadmap?

We have the same problem with one of our users who also wants to calculate a sum for fields that are calculated from a formula.
Would be nice if you could give me an update about the status.

Thanks!

**Comment by adrianszymanski89** [comment]
Hi @julianaiglsdorfer 

Thank you for contacting us. We have this issue reported internally. I'll update you as soon as we have the fix done.

**Reply by julianaiglsdorfer** [reply]
Thank you a lot for your effort. Do you have an approximate implementation date or time frame?

**Reply by adrianszymanski89** [reply]
Hi @julianaiglsdorfer 

Not yet. I will let you know as soon as we have it fixed.

**Reply by julianaiglsdorfer** [reply]
Hello @adrianszymanski89 do you have some information for me about this topic?

Are there any efforts to fix this issue?

Thanks in advance!

**Comment by sequba** [comment]
@julianaiglsdorfer, this is a feature request for the Handsontable product and should be discussed with their team within [Handsontable repository](https://github.com/handsontable/handsontable/issues).

However, it seems this feature may already have been shipped in Handsontable 18.0.0 via [this pull request](https://github.com/handsontable/handsontable/pull/12489). @adrianspdev can you confirm that?

**Reply by adrianspdev** [reply]
Confirmed. This was fixed in Handsontable 18.0.0 via https://github.com/handsontable/handsontable/pull/12489 (fixes https://github.com/handsontable/handsontable/issues/10980).

Before 18.0.0, columnSummary read cell values from the source data, so a cell containing =A1+B1 contributed the raw formula string instead of its calculated result and was skipped from the calculation. Since 18.0.0:

- columnSummary reads the value calculated by the formulas plugin, so sum, min, max, average, and count include formula cells.
- Summaries are refreshed automatically when HyperFormula recalculates a referenced cell, so they stay in sync after edits.

Changelog entry: https://github.com/handsontable/handsontable/blob/master/CHANGELOG.md#1800---2026-06-30

To get this, upgrade handsontable to 18.0.0 or later. If you still see the issue on 18.x, please open an issue in the Handsontable repository with a reproducible demo and we will take a look.

**Comment by sequba** [comment]
Thanks @adrianspdev. I'm closing this discussion.
